feat: Introduce Modulereleasemeta CRD (#1907)
kyma-project/lifecycle-manager: docs/contributor/resources/05-modulereleasemeta.md

# ModuleReleaseMeta 

The `modulereleasemetas.operator.kyma-project.io` Custom Resource Definition (CRD) defines the structure and format used to configure the ModuleReleaseMeta resource.

The ModuleReleaseMeta custom resource (CR) describes the channel-version pairs for modules. Each ModuleReleaseMeta represents one module and defines the available channels for this module along with the version that is currently assigned to the channel.

To get the latest CRD in the YAML format, run the following command:

```bash
kubectl get crd modulereleasemetas.operator.kyma-project.io -o yaml
```

## Configuration

### **.spec.moduleName**

The **moduleName** defines the name of the module for which the channel assignments are listed.

### **.spec.channels**

The **channels** define each module channel with its corresponding version. Each channel can only have one version assigned.
See the following example:

```yaml

> spec:
>   moduleName: keda
>   channels:
>     - channel: regular
>       version: 1.0.0
>     - channel: experimental
>       version: 1.1.0

Reviewer: ```suggestion
spec:
  moduleName: keda
  channels:
    - channel: regular
      version: 1.0.0
    - channel: fast
      version: 1.0.0
    - channel: experimental
      version: 1.1.0
```

related to above comment, to make it clear that the same version may appear in multiple channels.

# ModuleReleaseMeta Custom Resource

`modulereleasemetas.operator.kyma-project.io` Custom Resource Definition (CRD) defines the structure and format used to configure the ModuleReleaseMeta resource.

> The ModuleReleaseMeta custom resource is used to represent the version-channel pairs for modules. Each item represents

Author: yes so each module has the channel-version pairs defined in a ModuleReleaseMeta CR.

> The ModuleReleaseMeta custom resource is used to represent the version-channel pairs for modules. Each item represents
> a module version along with its assigned channel.

Reviewer: ```suggestion
The ModuleReleaseMeta custom resource (CR) is used to represent the version-channel pairs for modules. Each ModuleReleaseMeta represents one module and defines the available channels for this module along with the version that is currently assigned to the channel.
```

To get the latest CRD in the YAML format, run the following command:

```bash
kubectl get crd modulereleasemetas.operator.kyma-project.io -o yaml
```

## Configuration

### **.spec.moduleName**

> This parameter defines the name of the module for which the channel assignments are.

Reviewer: ```suggestion
The **moduleName** defines the name of the module for which the channel assignments are for.
```

### **.spec.channels**

> This parameter defines each channel with its corresponding version for the module. Each channel can only have one version assigned.
> An example could be:

Reviewer: ```suggestion
The **channels** define each channel of the module along with its corresponding version. Each channel can only have one version assigned.
An example is:
```

# ModuleReleaseMeta 

The `modulereleasemetas.operator.kyma-project.io` Custom Resource Definition (CRD) defines the structure and format used to configure the ModuleReleaseMeta resource.

> The ModuleReleaseMeta custom resource (CR) describes the channel-version pairs for modules. Each ModuleReleaseMeta represents one module and defines the available channels for this module along with the version that is currently assigned to the channel.

Reviewer 2: ```suggestion
The ModuleReleaseMeta custom resource (CR) describes the channel-version pairs for modules. Each ModuleReleaseMeta resource represents one module and defines the available channels for this module along with the version that is currently assigned to each of the channels.
```
